File: problems/can-scramble/can-scramble.php
<?php
function sorted_chars(string $input): array {
    $chars = str_split($input);
    sort($chars);
    return $chars;
}
function can_scramble(string $input, string $output): bool {
    return strlen($input) === strlen($output) && sorted_chars($input) === sorted_chars($output);
}
assert(can_scramble("abc", "abc") === true);
assert(can_scramble("abc", "cba") === true);
assert(can_scramble("a", "aaaaaaaaaa") === false); // Word length does not match
assert(can_scramble("abc", "cbad") === false);     // Word length does not match
assert(can_scramble("aab", "bba") === false);      // Word length matches, character count not
assert(can_scramble("ac", "bb") === false);        // Word length matches, wrong characters

File: problems/can-scramble/CanScramble.java
import java.util.HashMap;
import java.util.Map;

public class CanScramble {
    public static void main(String[] args) {
        assert canScramble("abc", "cba");
        assert !canScramble("ac", "bb");
        assert !canScramble("aab", "bba");
    }

    private static boolean canScramble(final String lhs, final String rhs) {
        return lhs.length() == rhs.length()
            && scrambleMap(lhs).equals(scrambleMap(rhs));
    }

    private static Map<Character, Integer> scrambleMap(final String s) {
        final Map<Character, Integer> m = new HashMap<>();
        for (final char c : s.toCharArray()) {
            m.merge(c, 1, Integer::sum);
        }
        return m;
    }
}
